dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,519 @@
+                # find input array
+                src_node = find_input_arraynode(dfg, edge)
+                # dispatch code gen (copy_memory)
+                self.dispatcher.dispatch_copy(src_node, node, edge, sdfg, dfg, state_id, function_stream,
+                                              callsite_stream)
+            # generate code to handle data output from the tasklet
+            for edge in dfg.out_edges(node):
+                # find output array
+                dst_node = find_output_arraynode(dfg, edge)
+                # dispatch code gen (define_out_memlet)
+                self.dispatcher.dispatch_output_definition(node, dst_node, edge, sdfg, dfg, state_id, function_stream,
+                                                           callsite_stream)
+            # generate tasklet code
+            self.unparse_tasklet(sdfg, dfg, state_id, node, function_stream, callsite_stream)
+        else:
+            raise RuntimeError(
+                "Only tasklets are handled here, not {}. This should have been filtered by the predicate".format(
+                    type(node)))
+
+    def copy_memory(self,
+                    sdfg: dace.SDFG,
+                    dfg: StateSubgraphView,
+                    state_id: int,
+                    src_node: nodes.Node,
+                    dst_node: nodes.Node,
+                    edge: MultiConnectorEdge,
+                    function_stream: CodeIOStream,
+                    callsite_stream: CodeIOStream):
+        """
+            Generate input/output memory copies from the array references to local variables (i.e. for the tasklet code).
+        """
+        if isinstance(edge.src, nodes.AccessNode) and isinstance(edge.dst, nodes.Tasklet):  # handle AccessNode->Tasklet
+            if isinstance(dst_node.in_connectors[edge.dst_conn], dace.pointer):  # pointer accessor
+                line: str = "{} {} = &{}[0];".format(dst_node.in_connectors[edge.dst_conn].ctype, edge.dst_conn,
+                                                     edge.src.data)
+            elif isinstance(dst_node.in_connectors[edge.dst_conn], dace.vector):  # vector accessor
+                line: str = "{} {} = *({} *)(&{}[0]);".format(dst_node.in_connectors[edge.dst_conn].ctype,
+                                                              edge.dst_conn,
+                                                              dst_node.in_connectors[edge.dst_conn].ctype,
+                                                              edge.src.data)
+            else:  # scalar accessor
+                line: str = "{}* {} = &{}[0];".format(dst_node.in_connectors[edge.dst_conn].ctype, edge.dst_conn,
+                                                      edge.src.data)
+        else:
+            raise RuntimeError("Not handling copy_memory case of type {} -> {}.".format(type(edge.src), type(edge.dst)))
+        # write accessor to file
+        callsite_stream.write(line)
```

Comment:
Why is this necessary instead of the regular CPU copy, it seems redundant? Can't we delegate this?

Reply:
Well, i didn't add the copy_dispatcher first, but that way the local variables that read/write from the function parameters didn't get created at all. Furthermore, i do have more flexibility by that, since i for example prefer to always use the pointers (instead of pointers only for non-scalar data like arrays).